### ISSUE TITLE
Track Ptrs in optimized code

### ISSUE THREAD

**Issue by emilydolson** [devosoft/Empirical#417]
As suggested in #110, now that `emp_always_assert` is implemented, it would be useful to use it in Ptr.h in place of standard `emp_assert`. That way it will work out of debug mode and make it possible to debug pointer issues that only happen in optimized code.

**Comment by mercere99** [comment]
Good call.  We should make sure to only use it in there code where EMP_TRACK_MEM is turned on.  Any asserts outside of that should be off when code is optimized (I don't think there are any, but it is important for optimized code to correctly be fully optimized)
